File: frontend/Models/CustomerOrder.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace frontend.Models
{
    public class CustomerOrderItem
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("customer_order_id")]
        public Guid CustomerOrderId { get; set; }

        [JsonPropertyName("item_number")]
        public int? ItemNumber { get; set; }

        [JsonPropertyName("test_name")]
        public string? TestName { get; set; }

        [JsonPropertyName("sample_type")]
        public string? SampleType { get; set; }

        [JsonPropertyName("test_count")]
        public int TestCount { get; set; } = 1;

        [JsonPropertyName("norm_method")]
        public string? NormMethod { get; set; }
    }

    public class CustomerOrder
    {
        [JsonPropertyName("id")]
        public Guid Id { get; set; }

        [JsonPropertyName("order_number")]
        public string OrderNumber { get; set; } = string.Empty;

        [JsonPropertyName("date")]
        public DateTime Date { get; set; }

        [JsonPropertyName("client_name")]
        public string ClientName { get; set; } = string.Empty;

        [JsonPropertyName("client_direction")]
        public string? ClientDirection { get; set; }

        [JsonPropertyName("client_ruc")]
        public string? ClientRuc { get; set; }

        [JsonPropertyName("client_dv")]
        public string? ClientDv { get; set; }

        [JsonPropertyName("client_phone")]
        public string? ClientPhone { get; set; }

        [JsonPropertyName("project_name")]
        public string? ProjectName { get; set; }

        [JsonPropertyName("project_location")]
        public string? ProjectLocation { get; set; }

        [JsonPropertyName("project_responsable")]
        public string? ProjectResponsable { get; set; }

        [JsonPropertyName("project_responsable_phone")]
        public string? ProjectResponsablePhone { get; set; }

        [JsonPropertyName("project_responsable_email")]
        public string? ProjectResponsableEmail { get; set; }

        [JsonPropertyName("observations")]
        public string? Observations { get; set; }

        [JsonPropertyName("attended_by")]
        public string? AttendedBy { get; set; }

        [JsonPropertyName("approved_quotation_number")]
        public string? ApprovedQuotationNumber { get; set; }

        [JsonPropertyName("created_by")]
        public string? CreatedBy { get; set; }

        [JsonPropertyName("items")]
        public List<CustomerOrderItem> Items { get; set; } = new();
    }

    public class CustomerOrderItemCreate
    {
        [JsonPropertyName("item_number")]
        public int? ItemNumber { get; set; }

        [JsonPropertyName("test_name")]
        public string? TestName { get; set; }

        [JsonPropertyName("sample_type")]
        public string? SampleType { get; set; }

        [JsonPropertyName("test_count")]
        public int TestCount { get; set; } = 1;

        [JsonPropertyName("norm_method")]
        public string? NormMethod { get; set; }
    }

    public class CustomerOrderCreate
    {
        [JsonPropertyName("date")]
        public DateTime Date { get; set; } = DateTime.Today;

        [JsonPropertyName("client_name")]
        public string ClientName { get; set; } = string.Empty;

        [JsonPropertyName("client_direction")]
        public string? ClientDirection { get; set; }

        [JsonPropertyName("client_ruc")]
        public string? ClientRuc { get; set; }

        [JsonPropertyName("client_dv")]
        public string? ClientDv { get; set; }

        [JsonPropertyName("client_phone")]
        public string? ClientPhone { get; set; }

        [JsonPropertyName("project_name")]
        public string? ProjectName { get; set; }

        [JsonPropertyName("project_location")]
        public string? ProjectLocation { get; set; }

        [JsonPropertyName("project_responsable")]
        public string? ProjectResponsable { get; set; }

        [JsonPropertyName("project_responsable_phone")]
        public string? ProjectResponsablePhone { get; set; }

        [JsonPropertyName("project_responsable_email")]
        public string? ProjectResponsableEmail { get; set; }

        [JsonPropertyName("observations")]
        public string? Observations { get; set; }

        [JsonPropertyName("attended_by")]
        public string? AttendedBy { get; set; }

        [JsonPropertyName("approved_quotation_number")]
        public string? ApprovedQuotationNumber { get; set; }

        [JsonPropertyName("items")]
        public List<CustomerOrderItemCreate> Items { get; set; } = new();
    }
}

File: backend/app/schemas/customer_order.py
from typing import Optional, List
from pydantic import BaseModel
import datetime
import uuid

# CustomerOrderItem Schemas
class CustomerOrderItemBase(BaseModel):
    item_number: Optional[int] = None
    test_name: Optional[str] = None
    sample_type: Optional[str] = None
    test_count: int = 1
    norm_method: Optional[str] = None

class CustomerOrderItemCreate(CustomerOrderItemBase):
    pass

class CustomerOrderItem(CustomerOrderItemBase):
    id: int
    customer_order_id: uuid.UUID

    model_config = {"from_attributes": True}

# CustomerOrder Schemas
class CustomerOrderBase(BaseModel):
    date: datetime.date = datetime.date.today()
    client_name: str
    client_direction: Optional[str] = None
    client_ruc: Optional[str] = None
    client_dv: Optional[str] = None
    client_phone: Optional[str] = None
    
    project_name: Optional[str] = None
    project_location: Optional[str] = None
    project_responsable: Optional[str] = None
    project_responsable_phone: Optional[str] = None
    project_responsable_email: Optional[str] = None
    
    observations: Optional[str] = None
    attended_by: Optional[str] = None
    approved_quotation_number: Optional[str] = None

class CustomerOrderCreate(CustomerOrderBase):
    items: List[CustomerOrderItemCreate] = []

class CustomerOrderUpdate(CustomerOrderBase):
    items: Optional[List[CustomerOrderItemCreate]] = None

class CustomerOrder(CustomerOrderBase):
    id: uuid.UUID
    order_number: str
    created_by: Optional[str] = None
    items: List[CustomerOrderItem] = []

    model_config = {"from_attributes": True}
